### PR TITLE
Exposed RenderingDevice to script API

### PR DESCRIPTION
**Pull request by reduz**
image

Also added an easier way to load native GLSL shaders.

image

Extras:

Had to fix no-cache for subresources in resource loader, it was not properly working, making shaders not properly reload.

### CODE REVIEW

File: editor/plugins/shader_file_editor_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+void ShaderFileEditor::_editor_settings_changed() {
+
+	if (is_visible_in_tree() && shader_file.is_valid()) {
+		_update_options();
+	}
+}
+
+void ShaderFileEditor::_bind_methods() {
+
+	//ClassDB::bind_method("_params_changed", &ShaderFileEditor::_params_changed);
```
Comment (clayjohn) [inline editor/plugins/shader_file_editor_plugin.cpp:192]: Should just remove i guess.

File: servers/rendering/rendering_device.h [hunk 2]
```diff
@@ -796,8 +813,8 @@ class RenderingDevice : public Object {
 			}
 		};
 
-		StencilOperationState stencil_operation_front;
-		StencilOperationState stencil_operation_back;
+		StencilOperationState front_op;
+		StencilOperationState back_op;
```
Comment (clayjohn) [inline servers/rendering/rendering_device.h:817]: Is there a particular reason for this rename. It seems worse than the original name to me.

**Review by clayjohn — CHANGES_REQUESTED** [review]
Code looks fine mostly. But you need to generate the documentation pages to go along with this. Run the engine with ``--doctool .`` and then commit the results (after discarding unrelated changes)

File: servers/rendering/rendering_device.h (hunk 2 above)
Comment (reduz) [reply servers/rendering/rendering_device.h:817]: yeah, the binder exposes the whole thing as a sub-variable, and the name ended up being too long. Also stencil_ is redundant in this context.

**Comment by reduz** [comment]
> Code looks fine mostly. But you need to generate the documentation pages to go along with this. Run the engine with `--doctool .` and then commit the results (after discarding unrelated changes)

The doc generator (or the checks at least) is broken and does not support referencing enums from foreign classes, so the pre-commit hook breaks. This needs to be fixed but seems outside the scope of my PR.

https://pastebin.com/LcfC4hm3

**Comment by reduz** [comment]
ok NOW it should be working.. lets wait until CI passes..

**Pull request merged**